trap/src/main/java/com/cursorinsight/trap/TrapManager.kt

```diff
@@ -113,35 +122,67 @@ class TrapManager internal constructor(
      */
     @Suppress("unused")
     fun halt(collector: TrapDatasource) {
-        val activity = currentActivity?.get()
-        if (activity != null) {
-            collectors[collector.hashCode()]?.stop(activity)
+        try {
+            val activity = currentActivity?.get()
+            if (activity != null) {
+                collectors[collector.hashCode()]?.stop(activity)
+            }
+            reporter.stop()
+        } catch (ex: Exception) {
+            Log.e(
+                TrapManager::class.simpleName,
+                "Stopping collector and reporter failed",
+                ex
+            )
         }
-        reporter.stop()
     }
 
     /**
      * Run all default collectors set in the configuration.
      */
     private fun runAll(activity: Activity) {
-        reporter.start()
-        for (collector in collectors.values) {
-            collector.start(activity)
+        try {
+            reporter.start()
+            for (collector in collectors.values) {
+                collector.start(activity)
+            }
+        } catch (ex: Exception) {
+            Log.e(
+                TrapManager::class.simpleName,
+                "Starting reporter and collectors failed",
```

Comment:
The subtle difference between the log message above in `run` and this one makes it difficult to find the reason **fast**.

I recommend something like this: "Starting reporter and collectors in runAll() failed."
And above: "Starting reporter and collector in run() failed."

Reply:
I've changed it